controller/src/main/java/io/pravega/controller/fault/ControllerClusterListener.java

```diff
@@ -74,18 +88,6 @@ protected void startUp() throws InterruptedException, Exception {
             log.info("Registering host {} with controller cluster", host);
             cluster.registerHost(host);
 
-            if (eventProcessorsOpt.isPresent()) {
```

Comment:
It is a good idea to have Sweeper interface, that has methods like `isReady`, `awaitReady`, `sweep(failedHost)`, `sweep(activeProcessSupplier)`, etc., so that ClusterListener can be initialized with the list of sweepers. 

We could then simplify the `startUp` method of `ClusterListener` as follows.
```
startup {

    cluster.addListener( HOST_REMOVED -> {
            sweepers.stream().parallel().map(sweeper -> {
                if (sweeper.isReady()) {sweeper.sweep(host.getHostId);}
            });
        }
    );

    sweepers.stream().parallel().map(sweeper -> {
        sweeper.awaitReady();
        sweeper.sweep(activeProcessesSupplier);
    });
}
```

Note that, there is no hostLost event lost with this way of organizing listener and initial sweeps.